bin/crew

```diff
@@ -983,11 +983,54 @@ def prepare_package(destdir)
 
     strip_dir destdir
 
+    # Use patchelf to set need paths for all binaries
+    # currently broken due to glibc issue on armv7l
+    unless ARCH == 'armv7l' || ARCH == 'aarch64'
+       patchelf_set_need_paths destdir
+    end
+
     # use upx on executables
     shrink_dir destdir
   end
 end
 
+def patchelf_set_need_paths(dir)
+  return unless @pkg.no_patchelf.to_s.empty?
+
+  Dir.chdir dir do
+    puts 'Running patchelf'.lightblue
+    abort('No Patchelf found!').lightred unless File.exist?("#{CREW_PREFIX}/bin/patchelf")
+
+    @execfiles = `find . -executable -type f ! \\( -name \"*.a\" \\) -exec sh -c \"file -i \'{}\' | grep -q \'executable; charset=binary\'\" \\; -exec ls -1i {} \\; | sort -u -n -s -k1,1 | awk '{print $2}'`.chomp
```

Comment:
How about checking the ELF file header directly instead of passing it to `file` and `grep` for ELFs?
```bash
bash -c "[[ \"\$(head -c4 {})\" == \$'\x7FELF' ]]"
```
(also it is much faster then `file+grep`):
```
chronos@localhost /usr/local/bin $ time bash -c "for f in *; do file -i \$f | grep -q 'executable; charset=binary'; done"

real    0m4.581s
user    0m7.016s
sys     0m1.639s
chronos@localhost /usr/local/bin $ time bash -c "for f in *; do [[ \"\$(head -c4 \$f)\" == \$'\x7FELF' ]]; done"

real    0m1.790s
user    0m0.790s
sys     0m1.103s
```

Reply:
I will have to see if this works...

Reply:
@supechicken I'm seeing this:
```
head: cannot open '{}' for reading: No such file or directory
bash: -c: line 1: conditional binary operator expected
bash: -c: line 1: syntax error near `$'ELF''
bash: -c: line 1: `[[  == $'ELF' ]]'
```

Reply:
with this line:
```
@execfiles = `find . -executable -type f ! \\( -name \"*.a\" \\) -exec bash -c \"[[ \"\$(head -c4 {})\" == \$'\x7FELF' ]]" \\; -exec ls -1i {} \\; | sort -u -n -s -k1,1 | awk '{print $2}'`.chomp
```

Reply:
I couldn't get the subshell to work. But this works:
```
    @execfiles = `find . -executable -type f ! \\( -name \"*.a\" \\) -exec head -c4 {} 2>/dev/null \\; -printf ' %i %p\\n'| grep ^.ELF | sort -u -n -s -k2,2 | awk '{print $3}'`.chomp

```